### PR TITLE
[chore]: v16 updated chanelog

### PR DESCRIPTION
**Pull request by stackman27**
Closes: #XXX

## What is the purpose of the change
updated changelog for v16 upgrade

### CODE REVIEW

**Comment by stackman27** [comment]
Please lmk if i missed anything. Happy to add it :))

**Review by mattverse — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## v16.0.0
+This release containts the following new modules and updates:
+- Concentrated Liquidity module (x/concentrated-liquidity). Concentrated liquidity is a novel Automated Market Maker (AMM) design introduced by Uniswap that allows for more efficient use of capital. The improvement is
```
Comment (AlpinYukseloglu) [inline CHANGELOG.md:45]:
```suggestion
- Concentrated Liquidity module (x/concentrated-liquidity). Concentrated liquidity is a gamechanging new pool model that will bring unprecedented capital efficiency to Osmosis. The initial version of our design is heavily influenced by Uniswap V3, although our implementation has a number of important differences.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
...
+achieved by providing liquidity in specific price ranges chosen by the user.
```
Comment (AlpinYukseloglu) [inline CHANGELOG.md:46]:
```suggestion
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
...
+- implement Cosmwasmpool module (x/cosmwasmpool). The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts.
```
Comment (AlpinYukseloglu) [inline CHANGELOG.md:47]:
```suggestion
- Implement Cosmwasmpool module (x/cosmwasmpool). The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts. 
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -75,6 +84,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
   * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
   * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
+  * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0
+  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) add amino proto annotations
```
Comment (AlpinYukseloglu) [inline CHANGELOG.md:88]:
```suggestion
  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -86,6 +98,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#4983](https://github.com/osmosis-labs/osmosis/pull/4983) Consume a gas when creating a new token using tokenfactory as a spam deterrence mechanism.
 * [#4951](https://github.com/osmosis-labs/osmosis/pull/4951) Implement pool liquidity query in pool manager, deprecate the one in gamm
 * [#5138](https://github.com/osmosis-labs/osmosis/pull/5138) Rename swap fee to spread factor. Broke PoolI interface. Did not break CLI, message, pool models, queries or events.
+* [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.
```
Comment (AlpinYukseloglu) [inline CHANGELOG.md:101]: Is this actually an API break, or should it be under misc improvements instead?

**Review by AlpinYukseloglu — APPROVED** [review]
Looks good but left a few comments (most important one is wording on CL release). Approving to not block merge but would like at least an ack on comments!

File: CHANGELOG.md [hunk 6]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
...
+## v16.0.0
+This release containts the following new modules and updates:
+- Concentrated Liquidity module (x/concentrated-liquidity). Concentrated liquidity is a novel Automated Market Maker (AMM) design introduced by Uniswap that allows for more efficient use of capital. The improvement is
+achieved by providing liquidity in specific price ranges chosen by the user.
+- implement Cosmwasmpool module (x/cosmwasmpool). The CosmWasm Pool Module is an extension for the Osmosis pools, aiming to create a custom module that allows users to create and manage liquidity pools backed by CosmWasm smart contracts. 
+- ProtoRev changes 
+  - Integrate with Concentrated Liquidity Pools. 
+  - Changes dev account payment from once per week to after every trade
+- Dependency Updates 
+  - bump wasmvm to v1.2 and add cosmwasm_1_2 capability
```
Comment (p0mvn) [inline CHANGELOG.md:52]: Could you please add that validators will have to update the wasm cache during the upgrade due to this:

https://osmosis-network.slack.com/archives/C027ELA290B/p1685870833358839?thread_ts=1685716114.561639&cid=C027ELA290B

File: CHANGELOG.md [hunk 7]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
...
+  - update async ICQ version 
+  - apply Cosmwasm Cherry security patch
```
Comment (p0mvn) [inline CHANGELOG.md:54]: Isn't cosmawsm cherry the same as "bump wasmvm to v1.2 and add cosmwasm_1_2 capability"?

File: CHANGELOG.md [hunk 8]
```diff
@@ -40,18 +40,27 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [Unreleased]
+## v16.0.0
```
Comment (p0mvn) [inline CHANGELOG.md:43]: Could you please also review our v16 upgrade handler and make sure that we specify all parameter updates in v16 here?

Comment (p0mvn) [reply CHANGELOG.md:43]: Please ping me on Slack once complete

Comment (stackman27) [reply CHANGELOG.md:43]: updated!

File: CHANGELOG.md [hunk 9]
```diff
@@ -40,52 +40,87 @@ All notable changes to this project will be documented in this file.
...
-  * [#5165](https://github.com/osmosis-labs/osmosis/pull/5165) Improve error message when fully exiting from a pool.
-  * [#5187](https://github.com/osmosis-labs/osmosis/pull/5187) Expand `IncentivizedPools` query to include internally incentivized CL pools.
-  * [#5239](https://github.com/osmosis-labs/osmosis/pull/5239) Implement `GetTotalPoolShares` public keeper function for GAMM.
-  * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
-  * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
-  * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
+* [#5356](https://github.com/osmosis-labs/osmosis/pull/5356) Fix wrong restHandler for ReplaceMigrationRecordsProposal
```
Comment (p0mvn) [inline CHANGELOG.md:98]: nit: I think indentation broke here

Comment (stackman27) [reply CHANGELOG.md]: oops fixed

File: CHANGELOG.md [hunk 10]
```diff
@@ -75,17 +115,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
   * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
   * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
+  * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0
```
Comment (p0mvn) [inline CHANGELOG.md:118]: This is duplicated under "Dependencies"

Comment (p0mvn) [reply CHANGELOG.md:118]: I think we should remove this

File: CHANGELOG.md [hunk 11]
```diff
@@ -75,17 +115,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
+  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add
```
Comment (p0mvn) [inline CHANGELOG.md:120]: Also duplicated
```suggestion
```

Comment (p0mvn) [inline CHANGELOG.md:120]: this as well
```suggestion
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -75,17 +115,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
   * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
   * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
+  * [#4783](https://github.com/osmosis-labs/osmosis/pull/4783) Update wasmd to 0.31.0
```
Comment (stackman27) [inline CHANGELOG.md:118]:
```suggestion
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -80,17 +127,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
```
Comment (p0mvn) [inline CHANGELOG.md:130]: duplicated
```suggestion
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -80,17 +127,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
+  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add
+  * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.
 
-### API breaks
 
-* [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod
```
Comment (p0mvn) [inline CHANGELOG.md:86]: I don't see this change present

**Review by p0mvn — APPROVED** [review]
Nice work. LGTM once 2 final comments are addressed

File: CHANGELOG.md [hunk 15]
```diff
@@ -80,17 +127,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   * [#5261](https://github.com/osmosis-labs/osmosis/pull/5261) Allows `UpdateFeeTokenProposal` to take in multiple fee tokens instead of just one.
   * [#5265](https://github.com/osmosis-labs/osmosis/pull/5265) Ensure a lock cannot point to multiple synthetic locks. Deprecates `SyntheticLockupsByLockupID` in favor of `SyntheticLockupByLockupID`.
   * [#4950] (https://github.com/osmosis-labs/osmosis/pull/4950) Add in/out tokens to Concentrated Liquidity's AfterConcentratedPoolSwap hook
+  * [#4629](https://github.com/osmosis-labs/osmosis/pull/4629) Add amino proto annotations
+  * [#4830](https://github.com/osmosis-labs/osmosis/pull/4830) Add gas cost when we AddToGaugeRewards, linearly increase with coins to add
+  * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.
```
Comment (stackman27) [inline CHANGELOG.md:132]:
```suggestion
  * [#5000](https://github.com/osmosis-labs/osmosis/pull/5000) osmomath.Power panics for base < 1 to temporarily restrict broken logic for such base.
  * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -80,17 +127,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+  * [#4336](https://github.com/osmosis-labs/osmosis/pull/4336) Move epochs module into its own go.mod
```
Comment (stackman27) [inline CHANGELOG.md:133]:
```suggestion
```

File: CHANGELOG.md (hunk 14 above)
Comment (stackman27) [reply CHANGELOG.md:86]: we move it here: https://github.com/osmosis-labs/osmosis/pull/5418/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR98

**Pull request merged**